src/compiler/checker.ts

```diff
@@ -16410,7 +16410,13 @@ namespace ts {
                 errorInfo = chainDiagnosticMessages(errorInfo, Diagnostics.Property_0_does_not_exist_on_type_1_Did_you_mean_2, declarationNameToString(propNode), typeToString(containingType), suggestion);
             }
             else {
-                errorInfo = chainDiagnosticMessages(errorInfo, Diagnostics.Property_0_does_not_exist_on_type_1, declarationNameToString(propNode), typeToString(containingType));
+                const promisedType = getPromisedTypeOfPromise(containingType);

```

Comment:
I would say that you should do the promise check before any sort of spelling checks because if a property exists on the awaited type, that's almost always a preferable suggestion.

Reply:
Will do